pkg/ottl/ottlfuncs/func_replace_match.go

```diff
@@ -29,11 +30,14 @@ func createReplaceMatchFunction[K any](_ ottl.FunctionContext, oArgs ottl.Argume
 		return nil, fmt.Errorf("ReplaceMatchFactory args must be of type *ReplaceMatchArguments[K]")
 	}
 
-	return replaceMatch(args.Target, args.Pattern, args.Replacement)
+	return replaceMatch(args.Target, args.Pattern, args.Replacement, args.Function)
 }
 
-func replaceMatch[K any](target ottl.GetSetter[K], pattern string, replacement ottl.StringGetter[K]) (ottl.ExprFunc[K], error) {
+func replaceMatch[K any](target ottl.GetSetter[K], pattern string, replacement ottl.StringGetter[K], fn ottl.Optional[ottl.FunctionGetter[K]]) (ottl.ExprFunc[K], error) {
 	glob, err := glob.Compile(pattern)
+	var replacementVal string
+	var replacementExpr ottl.Expr[K]
+	var replacementValRaw interface{}
```

Comment:
I think we should be able to declare these two on the lines where they're set.

Reply:
yes, made this change in the last push